services/ansibler/server/domain/usecases/update_no_proxy_envs.go

```diff
@@ -0,0 +1,179 @@
+package usecases
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/rs/zerolog/log"
+
+	commonUtils "github.com/berops/claudie/internal/utils"
+	"github.com/berops/claudie/proto/pb"
+	"github.com/berops/claudie/proto/pb/spec"
+	"github.com/berops/claudie/services/ansibler/server/utils"
+	"github.com/berops/claudie/services/ansibler/templates"
+)
+
+const (
+	defaulHttpProxyMode     = "default"
+	noProxyDefault          = "127.0.0.1/8,localhost,cluster.local,10.244.0.0/16,10.96.0.0/12" // 10.244.0.0/16 is kubeone's default PodCIDR and 10.96.0.0/12 is kubeone's default ServiceCIDR
+	noProxyPlaybookFilePath = "../../ansible-playbooks/update-noproxy-envs.yml"
+)
+
+type (
+	noProxyInventoryFileParameters struct {
+		K8sNodepools NodePools
+		ClusterID    string
+		NoProxy      string
+	}
+
+	NodePools struct {
+		Dynamic []*spec.NodePool
+		Static  []*spec.NodePool
+	}
+)
+
+func (u *Usecases) UpdateNoProxyEnvs(request *pb.UpdateNoProxyEnvsRequest) (*pb.UpdateNoProxyEnvsResponse, error) {
+	if request.Current == nil {
+		return &pb.UpdateNoProxyEnvsResponse{Current: request.Current, Desired: request.Desired}, nil
+	}
+
+	hasHetznerNodeFlag := hasHetznerNode(request.Desired.ClusterInfo)
+	httpProxyMode := commonUtils.GetEnvDefault("HTTP_PROXY_MODE", defaulHttpProxyMode)
+	// Changing NO_PROXY and no_proxy env variables is necessary only when
+	// HTTP Proxy mode isn't "off" and cluster has a Hetzner node or cluster is being build using HTTP proxy
+	if httpProxyMode == "off" || (httpProxyMode == "default" && !hasHetznerNodeFlag) {
+		return &pb.UpdateNoProxyEnvsResponse{Current: request.Current, Desired: request.Desired}, nil
+	}
+
+	nodesChangedFlag := nodesChanged(request.Current.ClusterInfo, request.Desired.ClusterInfo)
+	// At least one node in the cluster has to be changed to perform the update of NO_PROXY and no_proxy env variables.
+	if !nodesChangedFlag {
+		return &pb.UpdateNoProxyEnvsResponse{Current: request.Current, Desired: request.Desired}, nil
+	}
```

Comment:
I would remove this check, and just always update the noproxy envs, The workflow for building a cluster is run when you either add or delete nodes to the k8s cluster. (or also Lbs but in this case the redundancy wouldn't matter)